agent/consul/state/catalog_events.go

```diff
@@ -0,0 +1,447 @@
+package state
+
+import (
+	"github.com/hashicorp/consul/agent/consul/stream"
+	"github.com/hashicorp/consul/agent/structs"
+	memdb "github.com/hashicorp/go-memdb"
+)
+
+type changeOp int
+
+const (
+	OpDelete changeOp = iota
+	OpCreate
+	OpUpdate
```

Comment:
OpUpdate is not being used. If I am understanding how this works correctly, an update is communicated as 2 events, a delete followed by a create.